packages/datatrak-web/src/views/AccountSettingsPage/RequestCountryAccessSection/RequestCountryAccessForm.tsx

```diff
@@ -0,0 +1,244 @@
+  display: block flex;
+  flex-direction: column;
+  gap: 1.25rem;
+`;
+
+// Usage of this component has inline styling. See there for explanation.
+const StyledFormInput = styled(FormInput)`
+  flex-grow: 1;
+  margin: 0;
+
+  .MuiInputBase-root {
+    align-items: start;
+    block-size: 100%;
+    max-block-size: 100%;
+  }
+
+  .MuiInputBase-input {
+    box-sizing: border-box;
+  }
+`;
+
+interface RequestCountryAccessFormFields {
+  countryIds: Country['id'][];
+  reasonForAccess?: string;
+}
+
+export const RequestCountryAccessForm = () => {
+  const { project } = useCurrentUser();
+  const projectCode = project?.code;
+
+  const queryResult = useCountryAccessList();
+  const countries: TupaiaWebCountryAccessListRequest.ResBody = queryResult.data ?? [];
+  const accessListIsLoading = queryResult.isLoading;
+
+  const { mutate: requestCountryAccess, isLoading: requestIsLoading } = useRequestProjectAccess({
+    onError: error =>
+      errorToast(error?.message ?? 'Sorry, couldn’t submit your request. Please try again'),
+    onSettled: () => reset(),
+    onSuccess: response => {
+      successToast(response.message);
+      rerenderCountryChecklist();
+    },
+  });
+
+  /**
+   * When React Hook Form resets the form state, boxes get correctly unchecked, but the MUI
+   * component fails to reflect this until its next render (usually when the user interacts with it,
+   * which makes the checkbox feel broken). setCountryCheckList serves merely as an "automatic
+   * kicking machine" to force a re-render.
+   */
+  const [countryChecklistKey, setCountryChecklistKey] = useState(Date.now());
+
+  /**
+   * HACK: Force a fresh render of the country checklist. Use when a MUI checkbox appears checked,
+   * but is actually unchecked (or vice versa).
+   */
+  const rerenderCountryChecklist = () => setCountryChecklistKey(Date.now());
+
+  const sizeClassIsMdOrLarger = useMediaQuery(theme.breakpoints.up('sm'));
+
+  const formContext = useForm<RequestCountryAccessFormFields>({
+    defaultValues: {
+      countryIds: [],
+      reasonForAccess: '',
+    } as RequestCountryAccessFormFields,
+    mode: 'onChange',
+  });
+  const {
+    formState: { isSubmitting, isValidating, isValid },
+    handleSubmit,
+    register,
+    reset,
+  } = formContext;
+
+  const applicableCountries =
+    countries?.filter((country: TupaiaWebCountryAccessListRequest.CountryAccess) =>
+      project?.names?.includes(country.name),
+    ) ?? [];
+
+  const formIsNotSubmissible =
+    !project || isValidating || !isValid || isSubmitting || accessListIsLoading || requestIsLoading;
+
+  function onSubmit(formData: RequestCountryAccessFormFields) {
+    requestCountryAccess({
+      entityIds: formData.countryIds,
+      message: formData.reasonForAccess,
+      projectCode: projectCode,
+    });
+  }
+
+  return (
+    <StyledForm formContext={formContext} onSubmit={handleSubmit(onSubmit)}>
+      <StyledFieldset>
+        <CountryChecklistWrapper>
```

Comment:
I'd be tempted to split the checklist out into it's own component to reduce the amount of logic in this file